### PR TITLE
chore(flow): update flow-bin to version 0.112.0

### PR DESCRIPTION
**Pull request by chasestarr**
updates based on flow-bin changes described here: https://medium.com/flow-type/spreads-common-errors-fixes-9701012e9d58

### CODE REVIEW

File: documentation-site/examples/select/label.js [hunk 1]
```diff
@@ -6,13 +6,13 @@ import type {ValueT} from 'baseui/select';
 
 const ColorSwatch = styled<{$color: string}>('div', props => {
   return {
+    ...props.$theme.borders.border400,
```
Comment (tajo) [inline documentation-site/examples/select/label.js]: wonder if we should start turning everything into exact objects instead

Comment (chasestarr) [reply documentation-site/examples/select/label.js]: 👍 updated

**Comment by gergelyke** [comment]
do we want to keep this PR alive? it seems there are a ton of merge conflcits

**Comment by tajo** [comment]
> do we want to keep this PR alive? it seems there are a ton of merge conflcits

I am going to close it. There are probably two better options now:

A. Keep using the same version of flow internally and make changes only to satisfy our consumers (seems much easier).
B. Explore the TypeScript path.

**Pull request closed without merging**